packages/stream_chat_flutter/lib/src/message_widget/reactions/reaction_picker_icon_list.dart

```diff
@@ -0,0 +1,202 @@
+import 'package:collection/collection.dart';
+import 'package:ezanimation/ezanimation.dart';
+import 'package:flutter/material.dart';
+import 'package:stream_chat_flutter/src/misc/reaction_icon.dart';
+import 'package:stream_chat_flutter_core/stream_chat_flutter_core.dart';
+
+/// {@template onReactionPressed}
+/// Callback called when a reaction icon is pressed.
+/// {@endtemplate}
+typedef OnReactionPicked = void Function(Reaction reaction);
+
+/// {@template reactionPickerIconList}
+/// A widget that displays a list of reactionIcons that can be picked by a user.
+///
+/// This widget shows a row of reaction icons with animated entry. When a user
+/// taps on a reaction icon, the [onReactionPicked] callback is invoked with the
+/// selected reaction.
+///
+/// The reactions displayed are configured via [reactionIcons] and the widget
+/// tracks which reactions the current user has already added to the [message].
+/// {@endtemplate}
+class ReactionPickerIconList extends StatefulWidget {
+  /// {@macro reactionPickerIconList}
+  const ReactionPickerIconList({
+    super.key,
+    required this.message,
+    required this.reactionIcons,
+    this.onReactionPicked,
+  });
+
+  /// The message to display reactions for.
+  final Message message;
+
+  /// The list of available reaction icons.
+  final List<StreamReactionIcon> reactionIcons;
+
+  /// {@macro onReactionPressed}
+  final OnReactionPicked? onReactionPicked;
+
+  @override
+  State<ReactionPickerIconList> createState() => _ReactionPickerIconListState();
+}
+
+class _ReactionPickerIconListState extends State<ReactionPickerIconList> {
+  List<EzAnimation> iconAnimations = [];
+
+  void triggerAnimations() async {
+    for (final animation in iconAnimations) {
+      if (mounted) animation.start();
+      // Add a small delay between the start of each animation.
+      await Future.delayed(const Duration(milliseconds: 100));
+    }
+  }
+
+  void dismissAnimations() {
+    for (final animation in iconAnimations) {
+      animation.stop();
+    }
+  }
+
+  void disposeAnimations() {
+    for (final animation in iconAnimations) {
+      animation.dispose();
+    }
+  }
```

Comment:
More of a general comment, but I would make methods like these private. No need to have them public, even though the State class is already private. It's good practice to only make things public that should be public, so we don't have people use them when we don't expect it.